src/accelerate/utils/other.py

```diff
@@ -86,6 +89,23 @@ def extract_model_from_parallel(model, keep_fp32_wrapper: bool = True):
     while isinstance(model, options):
         model = model.module
 
+    if recursive:
+        # This is needed in cases such as using FSDPv2 on XLA
```

Comment:
Is there a way to detect that we are using fsdpv2 or xla , so that we don't have to set recursive. cc @shub-kris

Reply:
There are, but accelerate doesn’t support it yet and it’s experimental. So for now a generic approach, as it’s a single location in the Trainer